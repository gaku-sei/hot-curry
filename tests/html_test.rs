use anyhow::Result;
use insta::assert_snapshot;
use std::{fs::File, io::Read};

use hot_curry::resolve_html_assets;

fn open_html_input() -> Result<Vec<u8>> {
    let mut file = File::open("./tests/resources/index.html")?;

    let mut html = Vec::new();

    file.read_to_end(&mut html)?;

    Ok(html)
}

#[test]
fn it_prepends_base_url_to_relative_paths() -> Result<()> {
    let base_url = "https://fake.com/static/";

    let html = open_html_input()?;

    let new_html = resolve_html_assets(base_url, &mut html.as_slice())?;

    assert_snapshot!(new_html);

    Ok(())
}
