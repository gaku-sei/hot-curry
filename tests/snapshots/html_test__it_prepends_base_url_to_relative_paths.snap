---
source: tests/html_test.rs
expression: new_html

---
<html lang="en"><head>
    <meta charset="UTF-8">
    <meta content="IE=edge" http-equiv="X-UA-Compatible">
    <meta content="width=device-width, initial-scale=1.0" name="viewport">
    <link href="https://fake.com/static/styles.css" rel="stylesheet">
    <link href="https://already-an-url/styles.css" rel="stylesheet">
    <script rel="stylesheet" src="https://already-an-url/index.js"></script>
    <script rel="stylesheet" src="https://fake.com/static/index.js"></script>
    <style>
      @import url("https://fonts.googleapis.com/css2?family=Lato:wght@300&display=swap");

      body {
        font-family: "Lato", sans-serif;
      }

      @media print {
        .summary {
          break-inside: avoid-page;
        }
      }
    </style>
    <title>{{ basics.last_name }} {{ basics.first_name }}'s CV</title>
  </head>
  <body class="text-gray-800">
    <main class="flex flex-col md:flex-row">
      <img alt="already-an-url" src="https://already-an-url/image.png">
      <img alt="already-an-url" src="https://fake.com/static/image.png">
    </main>
  

</body></html>
